packages/nvidia_nat_test/src/nat/test/plugin.py

```
@@ -94,3 +104,91 @@
+@pytest.fixture(name="restore_environ")
+def restore_environ_fixture():
+    orig_vars = os.environ.copy()
+    yield os.environ
+
+    # Iterating over a copy of the keys as we will potentially be deleting keys in the loop
+    for key in list(os.environ.keys()):
+        orig_val = orig_vars.get(key)
+        if orig_val is not None:
+            os.environ[key] = orig_val
+        else:
+            del (os.environ[key])
```

Ruff (0.12.2)

No newline at end of file

Add trailing newline

(W292)